Update logstash-verifier image

msheiny opened the pull request:
So this PR does two things:
* updates the logstash-verifier image
* anddd updates a little bit of build logic

On the first issue, re-did the image completely based on an upstream openjdk image (MUCH easier build process with grsec), and did some re-engineering to try and make the image smaller (like grab upstream release package instead of compiling from source). I also tweaked the build entrypoint script so the image can be run as `logstash` user (🎉  without `root` ✨  ). 

So I also made some possible controversial changes to the build scripts. @conorsch need a ping here. FYI i've already built and pushed image `quay.io/freedomofpress/logstash-verifier:6.1.2` up and recorded that digest hash into the logstash-filters repo.

msheiny committed:
Utilize build-tags from container directories

Using the date as a tag doesnt provide us much metadata that is usually
important to have in a tag (like the version of the software
underneath). If we really want date build time, there are other ways to inspect
and filter on that.

msheiny committed:
Allow user to specify container to build

I'd rather use argparse here -- but I dont have time to
refactor now.

msheiny committed:
Big old refactor of logstash-verifier image

No longer needs to run as root, less dependencies installed, and did I
mention it no longer runs as root!?

msheiny committed:
Purge ansible retry files

conorsch commented on logstash-verifier/Dockerfile:
```diff
@@ -1,55 +1,35 @@
+    echo "$LS_VERIFIER_SHA256 ls-verifier.tar.gz" | sha256sum -c - && \
+    tar -C /usr/bin -xzf ls-verifier.tar.gz
+
+RUN chown logstash /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java && \
```
Wait, why does the `logstash` user need to _own_ the Java binary? It should be 755 by default.

msheiny replied:
Good question! Becauussee i need to change the pax flags on the java binary prior to logstash execution. In order to do that, the `logstash` user needs write permissions on it.

You might be saying... well wait a second.. i thought you already changed the pax flags earlier in the dockerfile using root permissions!? You are correct -- but unfortunately however docker handles the filesystem doesnt retain these filesystem flags and they effectively get wiped upon a new docker run.

tl;dr - in order to run the container as non-root, we need pax flags set at docker runtime by the logstash user

conorsch replied:
Thanks for clarifying, @msheiny, both here and in the appended comment. Let's dig more deeply on that separately, e.g. playing with `paxctl -c` versus `paxctl -C`.

conorsch reviewed (changes requested):
Build logic overhaul looks sound! Requesting clarification on a java `chown` that I'm not convinced needs to be there.

msheiny committed:
Comment for ownership change on java

I wasnt able to get past this issue and I've seen it before. I'm not
sure if its a storage provider issue or something having to do with the
layered file-systems.
